app/src/org/commcare/android/resource/analytics/InstallAttempts.java

```diff
@@ -0,0 +1,69 @@
+package org.commcare.android.resource.analytics;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Vector;
+
+/**
+ * Stores install attempt data for a given resource in the update table.
+ *
+ * @param <A> Type of data stored at each install attempt. Must be
+ *            instantiated with a class that implements Serializeable.
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+class InstallAttempts<A> implements Serializable {
```

Comment:
What is the purpose of using generics in this class? It looks like it's only getting used with type String right now, do you anticipate needing/wanting to use it with other types in the future?